wasm/jit/jit_arm64.go

```diff
@@ -327,16 +356,87 @@ func (c *arm64Compiler) compileCallIndirect(o *wazeroir.OperationCallIndirect) e
 	return fmt.Errorf("TODO: unsupported on arm64")
 }
 
+// compileDrop implements compiler.compileDrop for the arm64 architecture.
 func (c *arm64Compiler) compileDrop(o *wazeroir.OperationDrop) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	r := o.Range
+	if r == nil {
+		return nil
+	} else if r.Start == 0 {
+		// If the drop starts from the top, we just mark all the regiters
+		// used by drop targets unused.
+		for i := 0; i <= r.End; i++ {
+			if loc := c.locationStack.pop(); loc.onRegister() {
+				c.markRegisterUnused(loc.register)
+			}
+		}
+		return nil
+	}
+
+	// Save the non-drop target values.
+	liveValues := make([]*valueLocation, r.Start)
+	for i := 0; i < r.Start; i++ {
+		live := c.locationStack.pop()
+		liveValues[i] = live
+	}
+
+	// Then mark all registers used by drop tragets unused.
+	for i := 0; i < r.End-r.Start+1; i++ {
+		if loc := c.locationStack.pop(); loc.onRegister() {
+			c.markRegisterUnused(loc.register)
+		}
+	}
+
+	for i := range liveValues {
+		live := liveValues[len(liveValues)-1-i]
+		// If the value is on a memory, we have to move it to a register,
+		// otherwise the memory location is overriden by other values
+		// after this drop instructin.
+		if err := c.ensureOnGeneralPurposeRegister(live); err != nil {
+			return err
+		}
+		// Modify the stack location by pushing onto the location stack.
```

Comment:
It might be confusing that we are implying above "ensure it is on a GP register" and implying "it is on the stack" here :D This may be less confusing to people working on this all the time, though.

nit: modify is correct, but I think update is more telling. 
```suggestion
		// Update the stack location by pushing onto the location stack.
```

Reply:
So the reason why this is confusing is that we have two stacks -- one is valueLocationStack which we use at compilation time to track all the wasm values' location (memory stack, registers, or conditional register). Another is the stack in a memory at runtime. I must admit this is really confusing..

Reply:
https://github.com/tetratelabs/wazero/pull/214/commits/435623b84683600b96bdd9a13a6dde545bc6f9f9 I did my best only on this line, but I have to revisit all the comments later